src/main/groovy/com/wooga/gradle/io/ExecSpec.groovy

```diff
@@ -0,0 +1,162 @@
+     */
+    @InputDirectory
+    DirectoryProperty getExecutableDirectory() {
+        executableDirectory
+    }
+    private final DirectoryProperty executableDirectory = objects.directoryProperty()
+
+    void setExecutableDirectory(Provider<Directory> value) {
+        executableDirectory.set(value)
+    }
+
+    void setExecutableDirectory(File value) {
+        executableDirectory.set(value)
+    }
+
+    /**
+     * The name of the executable
+     */
+    @Input
+    Provider<String> getExecutableName() {
+        executableName
+    }
+    private final Property<String> executableName = objects.property(String)
+
+    void setExecutableName(Provider<String> value) {
+        executableName.set(value)
+    }
+
+    void setExecutableName(String value) {
+        executableName.set(value)
+    }
+
+    /**
+     * Sets the executable with the given file path.
+     * If it's a path, it will resolve the directory property, otherwise it will leave it empty
+     */
+    void setExecutableByPath(String path) {
+        def file = new File(path)
+        setExecutableByFile(file)
+    }
+
+    /**
+     * Sets the executable with the given file path.
+     * If it's a path, it will resolve the directory property, otherwise it will leave it empty
+     */
+    void setExecutableByPath(Provider<String> provider) {
+        setExecutableByFile(provider.map({new File(it)}))
+    }
+
+    /**
+     * Sets the executable by the provider
+     */
+    void setExecutableByFile(Provider<File> provider) {
+        executableName.set(provider.map({ it.name }))
+        executableDirectory.set(layout.dir(provider.map { it.parentFile }))
+    }
+
+    /**
+     * Sets the executable by the provider
+     */
+    void setExecutableByFile(File file) {
+        executableName.set(file.name)
+        executableDirectory.set(file.parentFile)
+    }
+
+    /**
+     * Sets the executable by the provider
+     */
+    void setExecutableByRegularFile(Provider<RegularFile> provider) {
+        setExecutableByPath(provider.map({ it.asFile.path }))
+    }
+
+    /**
+     * Sets the executable by the provider
+     */
+    void setExecutableByRegularFile(RegularFile file) {
+        setExecutableByFile(file.asFile)
+    }
+    /**
+     * To override the working directory when invoking the executable
+     * @return
+     */
+    @Input
+    @Optional
+    Property<String> getWorkingDirectory() {
+        workingDirectory
+    }
+
+    private final Property<String> workingDirectory = objects.property(String)
+
+    void setWorkingDirectory(Provider<String> value) {
+        workingDirectory.set(value)
+    }
+
+    /**
+     * Invokes the set executable
+     * @param configure An optional configuration step (for arguments, streams, etc)
+     * @return The result of the execution
+     */
+    ExecResult execute(@ClosureParams(value = FromString.class, options = ["org.gradle.process.ExecSpec"]) Closure configure) {
```

Comment:
as `Closure configure` is optional, you could default it as null or an empty closure so we have a `execute()` signature as well. Something like `execute(@ClosureParams.... Closure configure = {})`. Setting the default as a empty closure also removes the need for a null check on L156 as well.

Reply:
Yes, this is something I could do.

Reply:
Though at the moment I believe either approach is valid.